lib/Driver/FrontendUtil.cpp

```diff
@@ -57,6 +58,25 @@ bool swift::driver::getSingleFrontendInvocationFromDriverArguments(
   if (Diags.hadAnyError())
     return true;
 
+  if (ForceNoOutputs) {
+    // Clear existing output modes.
+    ArgList->eraseArg(options::OPT_modes_Group);
+    // Disable other output options that conflict with -typecheck.
```

Comment:
Nit: There's nothing really "conflicting" here—invoking with `-typecheck` is not supposed to disable these options, we rely on `-typecheck` working this way in many tests, and (e.g.) `-typecheck -emit-module-interface` is the correct way to emit a module interface without any other files. You're disabling them because you don't want any outputs, even secondary ones.

Reply:
Hmm, I guess it's not specifically the `-typecheck` flag that they're conflicting with, but typecheck being the only mode/action.  This comment is specific to the `-emit-*-path` options, since once you remove `-emit-module` (via removing the modes group), you get an error if you have `-emit-module-path`, which is why I was describing a conflict:

```
error: this mode does not support emitting modules
```

What about this wording: "Disable -emit-*path options that are invalid after removing -emit-* above" ?